drivers/can/Kconfig.mcp2515

```diff
@@ -20,34 +20,6 @@ config CAN_MCP2515_OSC_FREQ
 	help
 	  Specify the frequency of the oscillator connected to the MCP2515.
 
-config CAN_PROP_SEG
-	int "Prop_Seg"
-	default 2
-	range 1 8
```

Comment:
@karstenkoenig  I'm not sure if this is a good idea or not but as a user I have appreciated the information the range property Kconfig gives me (and stops me from doing bad things), is there any equivalent to check the device tree property settings?  The first solution that come to mind is build asserts.  There are runtime asserts in mcp2515_configure() but we could add build asserts for these CAN properties and possibly remove some runtime asserts.  Thoughts?

Reply:
Good idea, I converted the runtime asserts to buildtime asserts!